### DISCUSSION TITLE
Is there any unboxing avoid null check

### DISCUSSION THREAD

**Discussion by samanhappy** [checkstyle/checkstyle#13030] [Q&A] [answered]
like `f(id)` in below code block
```
public class Test {
  public static Integer id;

  public static void f(int i) {}

  public static void main(String[] args) {
    f(id);
  }
}
```

**Comment by romani** [comment, marked as answer]
Types are hard for our tool https://checkstyle.org/writingchecks.html#Limitations

In more complicated case we will not be able to get to know actual type of variable.
